### ISSUE TITLE
Running remote anvi'o interactive interfaces on my local computer

### ISSUE THREAD

**Issue by 473021677** [merenlab/anvio#1315]
Hi,
I am trying to install anvi'o on my Linux server. I have installed using below commands:
cd /data/data/yuanyang/softwares
conda create -n anvio-6.1 python=3.6
conda activate anvio-6.1
conda install -y -c conda-forge -c bioconda anvio=6.1
conda install -y diamond=0.9.14
But when I typed the command "anvi-self-test --suite mini", the browser didn’t pop-up. When I typed "anvi-self-test --version", it looked like this: 
```
Anvi'o version ...............................: esther (v6.1)
Profile DB version ...........................: 31
Contigs DB version ...........................: 14
Pan DB version ...............................: 13
Genome data storage version ..................: 6
Auxiliary data storage version ...............: 2
Structure DB version .........................: 1
```
And I tried to forward displays from the Linux server to my laptop according to the online tutorial "Running remote anvi'o interactive interfaces on my your computer". When I tried to allocate the port numbers, I couldn't find the "{path-to-anvio-virtual-env}/bin/activate" script on the Linux server. So I couldn't put the code below at the end of the "{path-to-anvio-virtual-env}/bin/activate" script.
[[ "$(whoami)" = "meren" ]] && export ANVIO_PORT=8080
[[ "$(whoami)" = "yuanyang" ]] && export ANVIO_PORT=22

Best regards,
Yuan

**Comment by 473021677** [comment]
Hi:
Thanks for your reply very much. The {conda env path} is "/data/data/yuanyang/softwares/bin". So I have created the path {conda env path}/etc/conda/activate.d as "/data/data/yuanyang/softwares/bin/etc/conda/activate.d". But I couldn't find the shell script "port_allocation.sh" for the anvio-6.1 directory on the Linux server. Sould I create the script? And I have created the script "port_allocation.sh". The file content is like:
#!/bin/bash

[[ "$(whoami)" = "yuanyang" ]] && export ANVIO_PORT=8080
I have put this script to "/data/data/yuanyang/softwares/bin/etc/conda/activate.d" and I tried to forward displays from the Linux server to my laptop according to the online tutorial "Running remote anvi'o interactive interfaces on my your computer". But when I allocated the port number, nothing shows when I typed "echo $ANVIO_PORT". I don't konw how to solve this problem. Could you help me? And how I should allocate the port number or maybe I could allocate the port number arbitrarily such as 8080, 8090 and other number? 

Best regards,
Yuan

**Comment by meren** [comment]
Yuan,

Make sure the script under /data/data/yuanyang/softwares/bin/etc/conda/activate.d has run permissions. Try,

```
chmod +x /data/data/yuanyang/softwares/bin/etc/conda/activate.d/port_allocation.sh
```

Add this into it:

```
export SIMPLE_TEST="worked"
```

Re-activate your environment, and see if this says worked:

```
echo $SIMPLE_TEST
```

If it doesn't, it means `port_allocation.sh` is not working. If it does, it means everything is working on your side.

Either ways, find a local bioinformatics guru and ask for their help since this is not an anvi'o issue and may require a better understanding of the local system, which is hard to do from here.


Best,

**Comment by 473021677** [comment]
Hi:
Thanks for your help very much. I have typed "chmod +x /data/data/yuanyang/softwares/bin/etc/conda/activate.d/port_allocation.sh" and "export SIMPLE_TEST="worked". When I typed "echo $SIMPLE_TEST", the "worked" showed up. This means everyting is working on my side. I will ask the bioinformatics guru for help. And I have another two questions to ask for you. The first one is that how I could determine the connection protocol just as shown "ozcan@barhal.mbl.edu"  for the alias "alias barhali="ssh -L 8090:localhost:8090 ozcan@barhal.mbl.edu | tee /dev/tty | python3 ~/.ssh/run_webbrowser.py" when I created an alias. When the user and host were yuanyang and 202.116.74.136, was the connection protocol yuanyang@202.116.74.136? The second question is whether I could substitute "127.0.0.1" for the line "line = line.replace("0.0.0.0", "127.0.0.1")" of the file "run_webbrowser.py" with "52.175.34.140" if my local computer IP address was "52.175.34.140"? Looking forward to your reply. Thanks.


Best regards,
Yuan
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: "A. Murat Eren"; 
Date: 2019年12月13日(星期五) 晚上9:46
To: "merenlab/anvio"; 
Cc: "473021677"; "Author"; 
Subject: Re: [merenlab/anvio] Running remote anvi'o interactive interfaces on my local computer (#1315)

&nbsp;

Yuan,
 
Make sure the script under /data/data/yuanyang/softwares/bin/etc/conda/activate.d has run permissions. Try,
 chmod +x /data/data/yuanyang/softwares/bin/etc/conda/activate.d/port_allocation.sh  
Add this into it:
 export SIMPLE_TEST="worked"  
Re-activate your environment, and see if this says worked:
 echo $SIMPLE_TEST  
If it doesn't, it means port_allocation.sh is not working. If it does, it means everything is working on your side.
 
Either ways, find a local bioinformatics guru and ask for their help since this is not an anvi'o issue and may require a better understanding of the local system, which is hard to do from here.
 
Best,
